### PR TITLE
Improved running transition through API

### PR DESCRIPTION
**Pull request by mkurek**
- use 'choices' param for ChoiceField (ex. callback function)
- return 400 in case of error, 201 when OK for sync transition and 202 for async
- additional tests for assigning new hostname through API

### CODE REVIEW

File: src/ralph/deployment/tests/test_transitions.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    DataCenterAssetFactory,
+    RackFactory
+)
+from ralph.lib.transitions.conf import TRANSITION_ORIGINAL_STATUS
+from ralph.lib.transitions.tests import TransitionTestCase
+from ralph.networks.tests.factories import (
+    NetworkEnvironmentFactory,
+    NetworkFactory
+)
+# from ralph.virtual.models import VirtualServer
```
Comment (szok) [inline src/ralph/deployment/tests/test_transitions.py]: Why commented?

Comment (mkurek) [inline src/ralph/deployment/tests/test_transitions.py]: for the same reason as below code - it was waiting for separated PR to be merged (#2481).

**Comment by szok** [comment]
:ok:

File: src/ralph/deployment/tests/test_transitions.py (hunk 1 above)
Comment (szok) [inline src/ralph/deployment/tests/test_transitions.py]: ok

**Comment by ghost** [comment]
:ok:

**Pull request merged**